### PR TITLE
Document Go job worker usage

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds documentation for the Go job worker, including for the new streaming job worker. It seems there was none at all prior to this, so I included some generic documentation based on the Java client's.

## When should this change go live?

- [x] This change is not yet live and should not be released until 8.4.0 release
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already (partially) available but undocumented
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/concepts/job-workers.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ Two streams are considered logically equivalent if they would both activate the
 - Have the same worker name
 - Have the same job activation timeout
 - Have the same fetch variables
-  :::
+:::
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply docs/components/concepts/job-workers.md:113]: Maybe I need a new line so it stops trying to fix it like this.

**Review by Zelldon — APPROVED** [review]
Thanks @npepinpe for writing this 👍🏼 Looks good to me, just smaller minor things.

File: docs/apis-tools/go-client/job-worker.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+- [Job worker reference](/components/concepts/job-workers.md)
+
+## The Go client's job worker
+
+The Go client provides a job worker that handles both polling and streaming for available jobs. This allows you to focus on writing code to handle the activated jobs.
+
+On `Open`, the job worker waits `PollInterval` milliseconds and then polls for `MaxJobsActive` jobs. It then continues with the following schedule:
+
+1. If a poll did not activate any jobs, it waits for `PollInterval` milliseconds and then polls for more jobs.
+2. If a poll activated jobs, the worker submits each job to the job handler.
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:19]: Yes. Do you think users should care about that, or it's more implementation detail?

File: docs/apis-tools/go-client/job-worker.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+3. Every time a job is handled, the worker checks whether the number of unhandled jobs have dropped below the configured `PollThreshold` (rounded up) of `MaxJobsActive`. The first time that happens, it will poll for more jobs.
+4. If a poll fails with an error response, a backoff strategy is applied. This strategy waits for the delay provided by the `BackoffSupplier` and polls for more jobs.
+
+For example, imagine you have 10 process instances and a single job worker configured with `maxJobsActive = 3`. The job worker will first pull three jobs and begin executing them. The threshold to poll for new jobs is 1 (30% of 3 rounded up). After two jobs have completed, the threshold is reached and the job worker will poll for up to 2 additional jobs. This process repeats until the jobs from all 10 process instances are completed.
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:23]: It's the default value, not a constant. I'll replace it. This was copied from Java where the threshold is hard coded I think.

File: docs/apis-tools/go-client/job-worker.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
+---
+id: job-worker
+title: "Job worker"
+description: "Let's take a deeper look at job workers to handle jobs."
+keywords: ["backpressure", "back-pressure", "back pressure"]
+---
+
+## Related resources
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:8]: Probably to some extent. I'll let our technical writing team decide on that :upside_down_face:

File: docs/apis-tools/go-client/job-worker.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+This ensures polling should not activate any new jobs, and the worker will back off and poll less often as long as it receives empty responses overtime.
+
+#### Backpressure
+
+To avoid your workers being overloaded with too many jobs, e.g. running out of memory, the Go job worker relies on the [built-in gRPC flow control mechanism](https://grpc.io/docs/guides/flow-control/). If streaming is enabled, this means the worker will never buffer more jobs than the configured `MaxJobsActive + Concurrency` parameters. For example, if `MaxJobsActive = 32` and `Concurrency = 10`, then your worker will only buffer at most 42 jobs concurrently. If this is already the case, and a 43rd job comes in, the streaming Go routine will block, thus signaling the gateway to stop sending more jobs.
+
+**If streaming is enabled, back pressure applies to both pushing and polling**. You can then use `MaxJobsActive` and `Concurrency` as a way to soft-bound the memory usage of your worker. For example, if your max message size is 4MB, and `MaxJobsActive = 22` and `Concurrency = 10`, then a single worker could use up to 128MB of memory in the worst case.
```
[the remark this answers is not in the document]
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:111]: Yeah. But the max message size does give you some upper bound, whereas the payload is variable sized, so if you can use max message size for the worst case scenario.

[the remark this answers is not in the document]
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md]: I rephrased it, hopefully it's better.

**Review by Zelldon — DISMISSED** [review]
🚀

**Comment by akeller** [comment]
@christinaausley looks like a review from you is the last step

**Review by christinaausley — APPROVED** [review]
Nice work on this 👍

**Comment by pepopowitz** [comment]
Two things I noticed with this PR: 

1. The merge timing between this & was such that one of the links is broken (from the new `go-client/job-worker` page, to `[gRPC technical error handling](/apis-tools/grpc.md#technical-error-handling)`). This is because #3150 reorganized the target of that link into a standalone page, and this branch didn't update before merging. None of this is a result of anyone doing something wrong, just bad timing! I'll fix this in a PR immediately, because it's preventing the docs from building.
2. The new `go-client/job-worker` page is not added to `sidebars.js`, so it's not appearing in the navigation, and when you navigate to it directly it shows no sidebar navigation. I suspect this is an oversight. I'll leave it up to @npepinpe or @christinaausley to resolve this!

**Comment by npepinpe** [comment]
Nice catch, I'll fix it in a follow up, sorry for missing that!

**Pull request merged**